portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +25,30 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> fruitFacts;
+
+  @Override
+  public void init() {
+    fruitFacts = new ArrayList<>();
+    fruitFacts.add("A strawberry is not an actual berry");
+    fruitFacts.add("Kiwis contain more Vitamin C than oranges");
+    fruitFacts.add("The pineapple is actually a berry");    
+  }
```

Comment:
A few things:

1. Let's prefer to use ImmutableLists (immutable means they can't be modified after constructing) when it makes sense to. In general, immutable objects are preferred when possible because they're generally safer. You can use the com.google.common.collect version: https://guava.dev/releases/23.0/api/docs/com/google/common/collect/ImmutableList.html.
In general, let's try to use this library for all container-like data structures (lists, maps, etc). Guava in general (Google's public version of Java that contains extra libraries like this one) should be utilized when possible. See https://guava.dev/ to learn more.

2. Nice, I like how you made this a class variable! Let's also make this a constant and inline the defintion by using the ImmutableList.of() method. So it should look something like:
private static final ImmutableList FRUIT_FACTS = ImmutableList.of("Strawberry ...", "Kiwi...", ....);

The reason it's good to use a class variable is separate the logic from the constants that don't change. If you ever just need to add another "hello" message, you don't need to sift through the logic code to find this variable. Likewise, it makes the actual business logic of the doGet method a lot cleaner and concise. You're already doing all of this which is great, but just thought I'd add this extra context as an FYI since I did for Kevin and Megan as well :)

Reply:
Oh btw, I think I missed some bracket formatting so in my example declaration in 2, it should be:

private static final ImmutableList\<String\> FRUIT_FACTS...

Still getting used to GitHub :\

Reply:
Got it! Thank you for sharing the links on ImmutableLists and for explaining their use! Same for class variables :) Please lmk if I implemented it incorrectly